sdk/python/feast/repo_config.py

```diff
@@ -37,6 +37,26 @@ class DatastoreOnlineStoreConfig(FeastBaseModel):
 OnlineStoreConfig = Union[DatastoreOnlineStoreConfig, SqliteOnlineStoreConfig]
 
 
+class FileOfflineStoreConfig(FeastBaseModel):
+    """ Offline store config for local (file-based) store """
+
+    type: Literal["file"] = "file"
+    """ Offline store type selector"""
+
+
+class BigqueryOfflineStoreConfig(FeastBaseModel):
+    """ Offline store config for GCP Bigquery """
+
+    type: Literal["bigquery"] = "bigquery"
+    """ Offline store type selector"""
+
+    entity_dataset_name: Optional[StrictStr] = "feast"
```

Comment:
What if we just had a catchall `dataset` for the offline store? I can already see the need to have temp tables when doing large joins (to avoid OOM issues with BigQuery). Ideally we wouldnt define two datasets in the config. I think its natural to have a single dataset dedicated to Feast tables.

Reply:
This `dataset` refers specifically to BigQuery's Dataset concept though. Other DWHes may have different concepts. Not sure how we'd generalize this. Instead I think each offline store implementation can define their own configuration that it can fully manage.

Reply:
Cool, I'll rename this to `dataset`.